analysis/analysis.py

```diff
@@ -282,15 +283,135 @@ def get_word_frequency_differences(student_data):
     return ordered_responses
 
 
-def run_analysis():
+def compute_median_view_time(student_data):
+    """
+     Given a list of student response dicts,
+    return the median time (across all users) spent reading the text
+
+    :param student_data: list, student response dicts
+    :return: float, median amount of time users spend reading the text
+    """
+    list_of_times = []
+    for row in student_data:
+        for view_time in row.get('views'):
+            list_of_times.append(view_time)
+    list_of_times.sort()
+    if len(list_of_times) == 0:
+        median_view_time = 0
+    else:
+        median_view_time = list_of_times[int(len(list_of_times) / 2)]
+    return median_view_time
+
+
+def compute_mean_response_length(student_data):
+    """
+    Given a list of student response dicts,
+    return the mean character length (across all users) of the response
+
+    :param student_data: list, student response dicts
+    :return: float, mean number of characters in the user's response
+    """
+
+    list_of_responses = []
+    for row in student_data:
+        list_of_responses.append(row.get("response"))
+    mean_response_length = 0
+    for response in list_of_responses:
+        mean_response_length += len(response)
+    return mean_response_length / len(list_of_responses)
+
+
+def run_time_analysis_functions(student_data):
     """
+    Runs analysis functions related to the time students took to read the passage
+    :param student_data: the data to analyze
     Runs the analytical method on the reading data
     :return: None
     """
+    median_view_time = compute_median_view_time(student_data)
+    total_view_time = compute_total_view_time(student_data)
+    mean_response_length = compute_mean_response_length(student_data)
+    print(f'The total view time of all students was {total_view_time}.')
+    print(f'The median view time of all students was {median_view_time}.')
+    print(f'The mean response length of all students was {mean_response_length} characters.')
+
+
+def description_has_relevant_words(story_meaning_description, relevant_words):
+    """
+    Determine if the user's description contains a word relevant to the story's meaning
+    :param story_meaning_description: The three word description of the story that the user supplied
+    :param relevant_words: a list of words which show an understanding of the story's meaning
+    :return True if the description contains one of the relevant words or relevant_words is empty.
+        False otherwise
+    """
+    if len(relevant_words) == 0:
+        return True
+
+    lowercase_relevant_words = list(map(lambda s: s.lower(), relevant_words))
+    words_used_in_description = story_meaning_description.lower().split(' ')
+
+    for word in lowercase_relevant_words:
+        if word.lower() in words_used_in_description:
+            return True
+    return False
+
+
+RELEVANT_WORDS_FILE_PATH = 'data/words_related_to_story.txt'
```

Comment:
This doesn't need to be at module scope -- just stick it inside the one function that uses it.